public/app/features/trails/DataTrail.test.tsx

```diff
@@ -459,4 +477,79 @@ describe('DataTrail', () => {
+
+  describe('OTel resources attributes', () => {
+    let trail: DataTrail;
+    const preTrailUrl =
+      '/trail?from=now-1h&to=now&var-ds=edwxqcebl0cg0c&var-deployment_environment=oteldemo01&var-otel_resources=k8s_cluster_name%7C%3D%7Cappo11ydev01&var-filters=&refresh=&metricPrefix=all&metricSearch=http&actionView=breakdown&var-groupby=$__all&metric=http_client_duration_milliseconds_bucket';
+
+    function getOtelDepEnvVar(trail: DataTrail) {
+      const variable = sceneGraph.lookupVariable(VAR_OTEL_DEPLOYMENT_ENV, trail);
+      if (variable instanceof CustomVariable) {
+        return variable;
+      }
+      throw new Error('getDepEnvVar failed');
+    }
+
+    function getOtelJoinQueryVar(trail: DataTrail) {
+      const variable = sceneGraph.lookupVariable(VAR_OTEL_JOIN_QUERY, trail);
+      if (variable instanceof ConstantVariable) {
+        return variable;
+      }
+      throw new Error('getDepEnvVar failed');
+    }
+
+    function getOtelResourcesVar(trail: DataTrail) {
+      const variable = sceneGraph.lookupVariable(VAR_OTEL_RESOURCES, trail);
+      if (variable instanceof AdHocFiltersVariable) {
+        return variable;
+      }
+      throw new Error('getOtelResourcesVar failed');
+    }
+
+    beforeEach(() => {
+      trail = new DataTrail({});
+      locationService.push(preTrailUrl);
+      activateFullSceneTree(trail);
+      getOtelResourcesVar(trail).setState({ filters: [{ key: 'service_name', operator: '=', value: 'adservice' }] });
+      getOtelDepEnvVar(trail).setState({
+        text: 'production',
+        value: 'production',
+        options: [{ value: 'production', label: 'production' }],
+      });
+    });
+
+    it('should start with hidden dep env variable', () => {
+      const depEnvVarHide = getOtelDepEnvVar(trail).state.hide;
+      expect(depEnvVarHide).toBe(VariableHide.hideVariable);
+    });
+
+    it('should start with hidden otel resources variable', () => {
+      const resourcesVarHide = getOtelResourcesVar(trail).state.hide;
+      expect(resourcesVarHide).toBe(VariableHide.hideVariable);
+    });
+
+    it('should start with hidden otel join query variable', () => {
+      const joinQueryVarHide = getOtelJoinQueryVar(trail).state.hide;
+      expect(joinQueryVarHide).toBe(VariableHide.hideVariable);
+    });
+
+    it('should add history step for when updating the otel resource variable', () => {
+      expect(trail.state.history.state.steps[2].type).toBe('resource');
+    });
+
+    it('Should have otel resource attribute selected as "service_name=adservice"', () => {
+      expect(getOtelResourcesVar(trail).state.filters[0].key).toBe('service_name');
+      expect(getOtelResourcesVar(trail).state.filters[0].value).toBe('adservice');
+    });
+
+    it('Should have deployment environment selected as "production"', () => {
+      expect(getOtelDepEnvVar(trail).getValue()).toBe('production');
+    });
+
+    // BUG: custom variable does not update variable dependency with setState
```

Comment:
are we tracking this in somewhere?

Reply:
I made an issue in Scenes that shows in our board and we are tracking it here https://github.com/orgs/grafana/projects/516/views/1?pane=issue&itemId=79759856